src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp

```diff
@@ -179,10 +179,15 @@ ReplicatedMergeTreePartCheckThread::MissingPartSearchResult ReplicatedMergeTreeP
                     found_part_with_the_same_min_block = true;
                     parts_found.push_back(part_on_replica);
                 }
+
                 if (part_on_replica_info.max_block == part_info.max_block)
                 {
                     found_part_with_the_same_max_block = true;
-                    parts_found.push_back(part_on_replica);
+
+                    /// If we are looking for part like partition_X_X_level we can add part
+                    /// partition_X_X_(level-1) two times, avoiding it
+                    if (parts_found.empty() || parts_found.back() != part_on_replica)
```

Comment:
Why do you think that it's always the last part in `parts_found`? Do we need a similar change for a part with the same min block? Looks like it still can generate something like `... as the missing part 1_331_332_1 ... Parts: 1_332_332_0, 1_331_331_0,  1_332_332_0` (the first replica had only `1_332_332_0` and the second one had both) or `... as the missing part 1_331_332_1 ... Parts: 1_331_331_0, 1_331_331_0,  1_332_332_0` (the first replica had only `1_331_331_0`). Consider just using std::[unordered_]set